### PR TITLE
Organize namespaces and add JSDoc grunt task + template

### LINKED ISSUE #75: Define all namespaces
Currently every single JS class, function or object seems namespaced somewhere under window.wp in a long chain of nested objects.

To properly test JSDoc all these namespaces should be added first so that all new documentation can properly make use of them. JSDoc is unfortunately not smart enough to realise that if the namespace `wp.many.many.many.nested.layers.of.namespaces` exists that this also implies the existence of a `wp.many`, a `wp.many.many` etc. namespaces.

I would suggest first adding all namespaces where they are defined so that all functions may properly document `@memberOf`.

### PR DESCRIPTION
**Pull request by herregroen**
Fixes #75.

### CODE REVIEW

File: src/wp-includes/js/customize-loader.js [hunk 1]
```diff
@@ -1,8 +1,5 @@
 /* global _wpCustomizeLoaderSettings, confirm */
-/*
- * Expose a public API that allows the customizer to be
- * loaded on any page.
- */
```
Comment (ntwb) [inline src/wp-includes/js/customize-loader.js]: Why is this removed? It was added in https://core.trac.wordpress.org/changeset/33709

See https://core.trac.wordpress.org/ticket/41682#comment:14

File: Gruntfile.js [hunk 2]
```diff
@@ -420,6 +420,14 @@ module.exports = function(grunt) {
 				}
 			}
 		},
+		jsdoc : {
+			dist : {
+				dest: "jsdoc",
```
Comment (ntwb) [inline Gruntfile.js]: Quotes need to use single quotes `'` please

File: Gruntfile.js [hunk 3]
```diff
@@ -420,6 +420,14 @@ module.exports = function(grunt) {
...
+				options: {
+					configure : "jsdoc.conf.json"
```
Comment (ntwb) [inline Gruntfile.js]: Quotes need to use single quotes `'` please

File: jsdoc.conf.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    "theme": "united",
+    "outputSourceFiles": true,
+    "outputSourcePath": true,
+    "linenums": true
+  },
+  "opts": {
+    "template": "./node_modules/ink-docstrap/template",
+    "recurse": true
+  }
+}
```
Comment (ntwb) [inline jsdoc.conf.json]: JSON files should use tabs for indentation please (the exception is `package.json`)

File: src/wp-includes/js/customize-loader.js (hunk 1 above)
Comment (herregroen) [reply src/wp-includes/js/customize-loader.js]: I've replied in the wordpress ticket. This was a subjective call based on feeling the `@namespace` tag conveys the same information. I've no issue with adding similar comments back in.

Comment (adamsilverstein) [reply src/wp-includes/js/customize-loader.js]: I would prefer all the comments be left in unless they effect the namespace parsing. That way, this patch remains smaller and more focused. We can address these other changes in a follow up ticket if needed.

**Comment by herregroen** [comment]
merged with: https://core.trac.wordpress.org/ticket/41682
